### PR TITLE
Add Muon CI job

### LINKED ISSUE #18: Support building with Muon buildsystem
[Muon buildsystem](https://sr.ht/~lattis/muon/) is the Meson-compatible implementation in pure C language. It supports the majority of the Meson features.

### PR DESCRIPTION
**Pull request by XVilka**
# DO NOT SQUASH

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

[Muon buildsystem](https://sr.ht/~lattis/muon/) is the Meson-compatible implementation in pure C language. It supports the majority of the Meson features.

**Test plan**

CI is green

**Closing issues**

Closes https://github.com/rizinorg/ideas/issues/18

### CODE REVIEW

File: .github/workflows/muon.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+jobs:
+  build:
+    name: ubuntu-muon
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Disable CMake module (unsupported by Muon)
+      run: |
+        sed -i "s/import\('cmake'\)/import\('cmake', required: false\)/" rizin/meson.build
```
Comment (ret2libc) [inline .github/workflows/muon.yml]:
```suggestion
        sed -i "s/import\('cmake'\)/import\('cmake', required: false\)/" meson.build
```
i think

**Comment by XVilka** [comment]
Limited by https://todo.sr.ht/~lattis/muon/62

**Comment by XVilka** [comment]
So many warnings, I wonder why it sets `-Wpedantic` by default...

**Comment by annacrombie** [comment]
@XVilka muon sets `-Dwarning_level=3` by default but you can add `'warning_level=1'` to the `project(default_options:)` if you want to keep the same behavior as meson.

**Comment by XVilka** [comment]
So it's a bit different from what I have on Fedora (https://github.com/rizinorg/ideas/issues/18#issuecomment-1174468850) but also boils down to libuv and missing `-fPIC`:
```
[1246/1906] linking subprojects/libuv-v1.40.0/libuv.so
FAILED: subprojects/libuv-v1.40.0/libuv.so 
cc  -o subprojects/libuv-v1.40.0/libuv.so subprojects/libuv-v1.40.0/libuv.so.p/src/fs-poll.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/idna.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/inet.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/random.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/strscpy.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/threadpool.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/timer.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/uv-common.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/uv-data-getter-setters.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/version.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/async.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/core.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/dl.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/fs.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/getaddrinfo.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/getnameinfo.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/loop-watcher.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/loop.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/pipe.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/poll.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/process.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/random-devurandom.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/signal.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/stream.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/tcp.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/thread.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/tty.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/udp.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/linux-core.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/linux-inotify.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/linux-syscalls.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/procfs-exepath.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/pthread-fixes.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/random-getrandom.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/random-sysctl-linux.c.o subprojects/libuv-v1.40.0/libuv.so.p/src/unix/proctitle.c.o -pthread -shared -Wl,-soname,libuv.so -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group /usr/lib/x86_64-linux-gnu/libm.so /usr/lib/x86_64-linux-gnu/libdl.so -Wl,--end-group
/usr/bin/ld: subprojects/libuv-v1.40.0/libuv.so.p/src/unix/core.c.o: relocation R_X86_64_PC32 against symbol `environ@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
```
https://github.com/rizinorg/rizin/runs/7216319976?check_suite_focus=true#step:8:1551

**Comment by annacrombie** [comment]
@XVilka not using `-fPIC` is a regression that I just [fixed](https://git.sr.ht/~lattis/muon/commit/14b72b45f706f910f1bb7cb4efc1a45cd7954e47).

**Comment by XVilka** [comment]
@annacrombie thank you very much! That wraps up the support 👍

**Comment by annacrombie** [comment]
@XVilka to install you need to run `muon -C build install`.  Same for tests, there is no `test` target in `build.ninja`, you need to run `muon -C build test`.

**Comment by XVilka** [comment]
```
warn subproject does not override dependency 'libuv'
fallback ['libuv'] failed for 'libuv'
warn dependency ['libuv'] not found
```
Looks like it prevents from `libuv.so` being linked/installed:
```
rizin -v
rizin: error while loading shared libraries: libuv.so: cannot open shared object file: No such file or directory
```

**Comment by annacrombie** [comment]
Investigating...

**Comment by annacrombie** [comment]
I think I found the problem, another recent regression :(.  It should be fixed by [this commit](https://git.sr.ht/~lattis/muon/commit/5cce316abcfabf0d93891e1235abf6927f7d4166).  The problem was that muon always tried to fallback for not found dependencies even if required was 'auto' or 'false'.  This meant that libuv got built without the subproject call that set `default_library=static`, and so instead of `libuv.a` it built and linked `libuv.so`.  The libuv library was set to install: false, so the resulting rizin exe was broken.

**Comment by ret2libc** [comment]
> @XVilka muon sets `-Dwarning_level=3` by default but you can add `'warning_level=1'` to the `project(default_options:)` if you want to keep the same behavior as meson.

Why does it do that?

**Comment by annacrombie** [comment]
@ret2libc It was a decision I made early on, I like it because I think it is better to default to more warnings, and then let the project decide to turn them off, rather than relying on people to remember to enable warnings.

**Comment by XVilka** [comment]
@annacrombie thanks, everything works now!

**Review by thestr4ng3r — APPROVED** [review]
Nice!!!

**Review by ret2libc — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Pull request merged**